client/actions/comments.js

```diff
@@ -41,6 +49,28 @@ const postData = (comment, getState, dispatch) => {
     });
 };
 
-export const addComment = comment => (dispatch, getState) => Promise.resolve().then(() => {
-  return postData(comment, getState, dispatch);
-});
+export const deleteComment = ({ id, field }) => (dispatch, getState) => {
+  const state = getState();
+  const basename = state.application.basename.replace(/\/edit?/, '');
+  return fetch(`${basename}/comment/${id}`, {
+    method: 'DELETE',
+    credentials: 'include'
+  })
+    .response
+    .then(response => {
+      return response.json()
+        .then(json => {
+          if (response.status > 399) {
+            const err = new Error(json.message || `Failed delete comment with status code: ${response.status}`);
+            err.status = response.status;
+            Object.assign(err, json);
+            throw err;
+          }
+        })
```

Comment:
Is there a case for abstracting some of this boilerplate?

Reply:
yes, absolutely. Is it the case that I was just being extremely lazy? Also yes.